src/meta/src/hummock/manager/mod.rs

```diff
@@ -1270,6 +1418,61 @@ where
             .await;
         }
 
+        // Warn of table_ids that is not found in expected compaction group.
+        // It indicates:
+        // 1. Either these table_ids are never registered to any compaction group. This is FATAL
+        // since compaction filter will remove these valid states incorrectly.
+        // 2. Or the owners of these table_ids have been dropped, but their stale states are still
+        // committed. This is OK since compaction filter will remove these stale states
+        // later.
+        let mut branch_sstables = vec![];
+        sstables.retain_mut(|(compaction_group_id, sst)| {
+            let res = match compaction_groups.get(compaction_group_id) {
+                Some(compaction_group) => {
+                    let mut is_valid = true;
+                    for table_id in sst
+                        .table_ids
+                        .iter()
+                        .filter(|t| !compaction_group.member_table_ids().contains(t))
+                    {
+                        is_valid = false;
+                        tracing::warn!(
+                            "table {} doesn't belong to expected compaction group {}",
+                            table_id,
+                            compaction_group_id
+                        );
+                    }
+                    is_valid
+                }
+                None => false,
+            };
+            if !res {
+                sst.divide_version += 1;
+                let mut branch_groups = HashMap::new();
+                for (group_id, group) in &compaction_groups {
+                    let mut match_ids = sst.get_table_ids().clone();
+                    match_ids.retain(|t| group.member_table_ids().contains(t));
+                    if !match_ids.is_empty() {
```

Comment:
If this branch is never hit, the sst is silently ignored aka not committed. If it happens, it should be regarded a bug?

Reply:
> If this branch is never hit, the sst is silently ignored aka not committed. If it happens, it should be regarded a bug?

moving main branch's `trace::warning` here?

Reply:
> If this branch is never hit, the sst is silently ignored aka not committed. If it happens, it should be regarded a bug?

added `trace::warning`

Reply:
> If this branch is never hit, the sst is silently ignored aka not committed. If it happens, it should be regarded a bug?

now in this situation, it will trigger `table {} in SST {} doesn't belong to expected compaction group {}` many times